content/blog/2016-09-01-interview-with-shuttlecloud.md

```diff
@@ -0,0 +1,114 @@
+      * System metrics → Scripts run by Ansible.
+      * Business metrics → Ducksboard and email reports.
+      * Blackbox metrics → Pingdom.
+  * No standard alert system. Each metric type had different alerts (email, push notification, and so on).
+  * Some business metrics had no alerts. These were reviewed manually.
+
+## Why did you decide to look at Prometheus?
+
+We analyzed several monitoring and alert systems. We were eager to get our hands dirty and check if the a solution would succeed or fail. The system we decided to put to the test was Prometheus, for the following reasons:
+
+   * First of all, you don’t have to define a fixed metric system to start working with it; metrics can be added or changed in the future. This provides valuable flexibility when you don’t know all of the metrics you want to monitor yet.
+   * If you know anything about Prometheus, you know that metrics can have labels that abstract us from the fact that different time series are considered. This, together with its query language, provided even more flexibility and a powerful tool. For example, we can have the same metric defined for different environments or projects and get a specific time series or aggregate certain metrics with the appropriate labels:
+      * `http_requests_total{job="my_super_app_1",environment="staging"}` - the time series corresponding to the staging environment for the app "my\_super\_app_1".
+      * `http_requests_total{job="my_super_app_1"}` - the time series for all environments for the app "my\_super\_app\_1".
+      * `http_requests_total{environment="staging"}` - the time series for all staging environments for all jobs.
+   * Prometheus supports a DNS service for service discovery. We happened to already  have an internal DNS service.
+   * There is no need to install any external services (unlike Sensu, for example, which needs a data-storage service like Redis and a message bus like RabbitMQ). This might not be a deal breaker, but it definitely makes the test easier to perform, deploy, and maintain.
+   * Prometheus is quite easy to install, as you only need to download an executable Go file. The docker container also works well and it is easy to start.
+
+## How do you use Prometheus?
+
+Initially we were only using some metrics provided out of the box by the [node_exporter](https://github.com/prometheus/node_exporter), including:
+
+   * hard drive usage.
+   * memory usage.
+   * if an instance is up or down.
+
+Our internal DNS service is integrated to be used for service discovery, so every new instance is automatically monitored.
+
+Some of the metrics we used, which were not provided by the node_exporter by default, were exported using the [node_exporter textfile collector](https://github.com/prometheus/node_exporter#textfile-collector) feature. The first alerts we declared on the Prometheus Alertmanager were mainly related to the operation metrics mentioned above.
```

Comment:
Ah, maybe "operation metrics" is right after all, if it's about their business operations...